app/Models/UserGroupEvent.php

```diff
@@ -34,46 +36,69 @@ class UserGroupEvent extends Model
         'hidden' => 'boolean',
     ];
 
-    public static function logGroupRename(?User $actor, Group $group, string $oldName, string $newName): self
+    public static function logGroupRename(?User $actor, Group $group, string $previousName, string $name): self
     {
-        return static::log($actor, static::GROUP_RENAME, $group, [
+        return static::log($actor, static::GROUP_RENAME, null, $group, [
             'details' => [
-                'old_name' => $oldName,
-                'new_name' => $newName,
+                'group_name' => $name,
+                'previous_group_name' => $previousName,
             ],
         ]);
     }
 
-    public static function logUserAdd(?User $actor, User $user, Group $group): self
+    public static function logUserAdd(?User $actor, User $user, Group $group, ?array $playmodes = null): self
     {
-        return static::log($actor, static::USER_ADD, $group, [
-            'user_id' => $user->getKey(),
+        return static::log($actor, static::USER_ADD, $user, $group, [
+            'details' => [
+                'playmodes' => $group->has_playmodes ? ($playmodes ?? []) : null,
```

Comment:
if the function receives `$userGroup` instead, this check can be removed.

Reply:
i think this is fine as-is just to keep all the methods having similar signature & requirements